jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-showcase/src/main/java/org/jbpm/workbench/cm/server/MockCaseManagementService.java

```diff
@@ -72,21 +72,26 @@
     private Map<String, List<CaseActionSummary>> caseActionMap = new HashMap<>();
     private List<CaseActionSummary> caseActionList = new ArrayList<>();
     private List<ProcessDefinitionSummary> processDefinitionList = emptyList();
+    private final ObjectMapper mapper = new ObjectMapper();
 
     @PostConstruct
     public void init() {
-        try {
-            final ObjectMapper mapper = new ObjectMapper();
-            final InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(CASE_DEFINITIONS_JSON);
-            caseDefinitionList = Arrays.asList(mapper.readValue(inputStream, CaseDefinitionSummary[].class));
-            caseMilestoneList = Arrays.asList(mapper.readValue(Thread.currentThread().getContextClassLoader().getResourceAsStream(CASE_MILESTONES_JSON), CaseMilestoneSummary[].class));
-            caseStageList = Arrays.asList(mapper.readValue(Thread.currentThread().getContextClassLoader().getResourceAsStream(CASE_STAGES_JSON), CaseStageSummary[].class));
-            caseActionList = Arrays.asList(mapper.readValue(Thread.currentThread().getContextClassLoader().getResourceAsStream(CASE_ACTIONS_JSON), CaseActionSummary[].class));
-            processDefinitionList = Arrays.asList(mapper.readValue(Thread.currentThread().getContextClassLoader().getResourceAsStream(PROCESS_DEFINITION_JSON), ProcessDefinitionSummary[].class));
-
+            caseDefinitionList = readJsonValues(CaseDefinitionSummary.class, CASE_DEFINITIONS_JSON);
+            caseMilestoneList = readJsonValues(CaseMilestoneSummary.class, CASE_MILESTONES_JSON);
+            caseStageList = readJsonValues(CaseStageSummary.class, CASE_STAGES_JSON);
+            caseActionList = readJsonValues(CaseActionSummary.class, CASE_ACTIONS_JSON);
+            processDefinitionList = readJsonValues(ProcessDefinitionSummary.class, PROCESS_DEFINITION_JSON);
             LOGGER.info("Loaded {} case definitions", caseDefinitionList.size());
+    }
+
+    private <T> List<T> readJsonValues(final Class<T> type, final String fileName){
+        try {
+            final InputStream inputStream = Thread.currentThread().getContextClassLoader().getResourceAsStream(fileName);
```

Comment:
FYI the stream is not properly closed here. If put inside `try ()`, this will be fixed.